### PR TITLE
DataFrame.loc now supports indexing with list

### LINKED ISSUE #1298: DataFrame.loc does not work with list of indices
If `i = [3, 5, 1]` and `df` is a pandas data frame this returns the numbered rows in the defined order: `df.loc[i]`. 

This is the expected behavior for `dask.dataframe.DataFrame` but if `df` is such an object then `df.loc[i]` does not work. Note that you _can_ give it a slice: `df.loc[3:6]` for example.

On a side note, although using numbers in the example above, `loc` is for labels and as such we could expect partitions made of indices (or multindices) of strings, datetimes, and other objects...

### PR DESCRIPTION
**Pull request by sinhrks**
closes #1298.

### CODE REVIEW

File: dask/dataframe/tests/test_indexing.py [hunk 1]
```diff
@@ -120,6 +126,20 @@ def test_loc2d():
...
+                      index=list('abcdefghijklmnopqrst'),
+                      columns=list('ABCDE'))
+    ddf = dd.from_pandas(df, 3)
+
+    assert_eq(ddf.loc['a', 'A'], df.loc[['a'], 'A'])
+    assert_eq(ddf.loc['a', ['A']], df.loc[['a'], ['A']])
+    assert_eq(ddf.loc['a':'o', 'A'], df.loc['a':'o', 'A'])
+    assert_eq(ddf.loc['a':'o', ['A']], df.loc['a':'o', ['A']])
+    assert_eq(ddf.loc[['n'], ['A']], df.loc[['n'], ['A']])
+    assert_eq(ddf.loc[['a', 'c', 'n'], ['A']], df.loc[['a', 'c', 'n'], ['A']])
```
Comment (mrocklin) [inline dask/dataframe/tests/test_indexing.py]: What happens if the row index list is not sorted?

**Comment by mrocklin** [comment]
Tests fail due to a conda issue.  This has been reported.  They say that there will probably be a bugfix release on Monday.

**Comment by jreback** [comment]
fyi fixing conda to 4.2.15 makes appveyor work again :)

**Comment by mrocklin** [comment]
Conda issues resolved.  There is a doctest failure in 2.7.

File: dask/dataframe/tests/test_indexing.py (hunk 1 above)
Comment (sinhrks) [inline dask/dataframe/tests/test_indexing.py]: it works, added some tests.

**Comment by mrocklin** [comment]
Looks good to me.  Merging tomorrow if no further comments.

**Pull request merged**